### PR TITLE
Parse REST big numbers as string

### LINKED ISSUE #35: getBlockchainInfo verificationprogress rounding issue
bitcoin-cli returns me "verificationprogress": 1
but bitcoin-core returns me 0.9999941262782213

Since this is the return of the library I have no way to fix that rounding issue.
I would expect 1 to be returned.
(I know this is a common javascript problem.)
Generally speaking, I would expect this library to delive 8 decimals number (8 decimals for satoshis). Maybe return numbers as strings and let the client interpret it like he wants it to (many libraries available).

### PR DESCRIPTION
**Pull request by pedrobranco**
This PR adds support for parsing big numbers as string to prevent losing precision due to Javascript number precision problem.

Fixes #35 .

### CODE REVIEW

**Comment by ruimarinho** [comment]
Found a little issue when testing this PR due to the timing of the constructor of RpcError. @pedrobranco can you submit a PR for this?

```diff
diff --git i/src/errors/rpc-error.js w/src/errors/rpc-error.js
index 19ac8eb..997fdb2 100755
--- i/src/errors/rpc-error.js
+++ w/src/errors/rpc-error.js
@@ -11,7 +11,7 @@ import StandardError from './standard-error';
  */

 export default class RpcError extends StandardError {
-  constructor(code, msg, props) {
+  constructor(code, msg, props = {}) {
     if (typeof code != 'number') {
       throw new TypeError(`Non-numeric HTTP code`);
     }
@@ -21,9 +21,9 @@ export default class RpcError extends StandardError {
       msg = null;
     }

+    props.code = code;
+
     super(msg || STATUS_CODES[code], props);
-
-    this.code = code;
   }

   get status() {
```

**Comment by pedrobranco** [comment]
> @pedrobranco I am unable to reproduce this issue and don't actually see the originally reported issue anymore (nor a test preventing a regression). I also fail to understand what exactly this PR does as there is no BigInt returned on the verificationprogress field, nor one stored as a string as it was supposed to. Can you clarify a bit the changes?

Actually, when syncing the blockchain in `mainnet` (or `testnet`) the value returned by `bitcoind` is a double, which will loss precision on its JS interpretation:
```js
> cli.getBlockchainInfo().then(data => console.log(data.verificationprogress)).catch(console.err)

0.01725915139476017
```

File: src/index.js [hunk 1]
```diff
@@ -123,7 +123,6 @@ class Client {
       return this.request.postAsync({
         auth: _.pickBy(this.auth, _.identity),
         body: JSON.stringify(body),
-        json: false,
```
[the remark this answers is not in the document]
Comment (pedrobranco) [reply src/index.js]: This removes the override of the value `json` already set in `this.request`.

File: src/index.js [hunk 2]
```diff
@@ -89,7 +89,7 @@ class Client {
     this.request = Promise.promisifyAll(request.defaults({
       agentOptions: this.agentOptions,
       baseUrl: `${this.ssl.enabled ? 'https' : 'http'}://${this.host}:${this.port}`,
-      json: true,
+      json: false,
```
[the remark this answers is not in the document]
Comment (pedrobranco) [reply src/index.js]: Sure.

File: src/parser.js [hunk 3]
```diff
@@ -82,7 +82,16 @@ export default class Parser {
     return batch;
   }
 
-  rest([response, body]) {
+  rest(extension, [response, body]) {
+    if (extension === 'json') {
```
Comment (ruimarinho) [inline src/parser.js]: This should be moved further down.

**Comment by ruimarinho** [comment]
@pedrobranco in that case, shouldn't that be returned as a string?

**Comment by pedrobranco** [comment]
> @pedrobranco in that case, shouldn't that be returned as a string?

JSONBitInt only converts numeric values that will lose precision when parsing as JS object.

File: test/index_test.js [hunk 4]
```diff
@@ -390,9 +390,10 @@ describe('Client', () => {
 
     describe('getTransactionByHash()', () => {
       it('should return a transaction binary-encoded if extension is `bin`', async () => {
-        const transaction = await client.getTransactionByHash('b4dd08f32be15d96b7166fd77afd18aece7480f72af6c9c7f9c5cbeb01e686fe', { extension: 'bin' });
+        const [transaction] = await client.listUnspent();
+        const bin = await client.getTransactionByHash(transaction.txid, { extension: 'bin' });
 
-        new Buffer(transaction, 'binary').toString('hex').should.endWith('206e6f7420666f756e640d0a');
+        new Buffer(bin, 'binary').toString('hex').should.endWith('fdfdfd36fd4efdfd00000000');
```
Comment (ruimarinho) [inline test/index_test.js]: Needs the binary test fix before merging this in.

**Comment by ruimarinho** [comment]
@pedrobranco: me and @joaopaulofonseca took a look at this PR and there are a couple of things we feel important to be part of a general improvement to response handling, in which this change is included. Therefor, please refer to https://github.com/ruimarinho/bitcoin-core/pull/50 for further progress on this.

**Pull request closed without merging**